### PR TITLE
Fixes #1077 SymbolTable not giving back DisplayNames

### LINKED ISSUE #1077: SymbolTable  not giving back DisplayNames
ReadOnlySymbolTable can enumerate NamedFormulaType, but that only gives logical name and leaves .DisplayName empty. 

https://github.com/microsoft/Power-Fx/blob/91833f01001ec79a8a9190c8e046cd1d96be7709/src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs#L254

Consider both variables added with display name but also rowscope over Records with display names. 

We use this for tooling and intellisense features that suggest display names. 

This should also let us enable this test:
https://github.com/microsoft/Power-Fx-Dataverse/blob/fb6d11c22f1b1491ab2887f3886d7b340eb5d544/src/PowerFx.Dataverse.Tests/PluginExecutionTests/PluginExecutionTests.cs#L59

### PR DESCRIPTION
**Pull request by jas-valgotar**
Fixes #1077

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs [hunk 1]
```diff
@@ -260,8 +260,9 @@ public IEnumerable<NamedFormulaType> SymbolNames
                 // GlobalSymbols are virtual, so we get derived behavior via that.
                 foreach (var kv in globals.GlobalSymbols)
                 {
-                    var type = FormulaType.Build(kv.Value.Type);
-                    yield return new NamedFormulaType(kv.Key, type);
+                    var type = FormulaType.Build(kv.Value.Type);
+                    var displayName = kv.Value.DisplayName != default ? kv.Value.DisplayName.Value : null;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs:264]: >kv.Value.DisplayName != default ? kv.Value.DisplayName.Value : null; [](http://example.com/codeflow?start=38&length=68)

Can we use ?. operator?

 kv.Value.DisplayName?.Value

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs:264]: Since DName is struct, this syntax wouldn't work.

**Pull request merged**